src/Npgsql/NpgsqlReadBuffer.cs

```diff
@@ -209,7 +211,7 @@ async Task EnsureLong()
                     {
                     // User requested the cancellation
                     case OperationCanceledException _ when (cancellationToken.IsCancellationRequested):
-                        throw Connector.Break(e);
+                        throw dontBreakOnCancellation ? e : Connector.Break(e);
```

Comment:
It's ironic that we got rid of dontBreakOnTimeouts, just to get this back :rofl:

Reply:
Your comment has forced me to check again, if it really was needed 😰 
But yeah, no other way to go from here.